cumulus/parachains/runtimes/assets/common/src/foreign_creators.rs

```diff
@@ -42,8 +42,10 @@ where
 		origin: RuntimeOrigin,
 		asset_location: &L,
 	) -> core::result::Result<Self::Success, RuntimeOrigin> {
+		tracing::trace!(target: "xcm::try_origin", ?origin, ?asset_location, "ForeignCreators");
 		let origin_location = EnsureXcm::<Everything, L>::try_origin(origin.clone())?;
 		if !IsForeign::contains(asset_location, &origin_location) {
+			tracing::debug!(target: "xcm::try_origin", ?asset_location, ?origin_location, "ForeignCreators: asset location not contained in origin location");
```

Comment:
nit: keep it short and sweet

```suggestion
			tracing::trace!(target: "xcm::try_origin", ?asset_location, ?origin_location, "ForeignCreators: no match");
```